onebeyond/react-form-builder
# Feat/Added error classes to Select and Input for customization

### Type:

- [ ] CI/CD: helm, docker & CI/CD adjustments.
- [x] feature: new capabilities.
- [ ] fix: bug, hotfix, etc.
- [ ] refactor: enhancements.
- [x] style: changes in styles.
- [ ] other: docs, tests.

### What's the focus of this PR:

- Adds the `error-input` and `error-select` classes to the Input and Select fields when they have errors. It's the same concept as the `error-message` class that we already had for the ErrorMessage elements, so we can customize the fields instead of only being able to customize the text that shows when there's an error.

### How to review this PR:

- Run the example project but don't fill anything in the form.
- In the Elements inspector, add a New Style Rule.
- Call the new rule `.error-input` and put the error styles you want to apply. For example:

<img width="1440" alt="image" src="https://github.com/onebeyond/react-form-builder/assets/23655224/4a8c5a06-3ba1-4fec-a2d1-c1cd26277a57">

- Add another rule called `.error-select` and put the error styles you want to apply. For example:

<img width="1440" alt="image" src="https://github.com/onebeyond/react-form-builder/assets/23655224/6d1bd977-0cf0-4c33-9728-16bfbe05d2b0">

- Click the Enter button to submit the form. The styles you gave to the `.error-input` and `.error-select` should be visible. Note that these error classes enable to customize the field and any other classes of the elements/classes below them, but it's harder to test from the inspector.

### Related work items

- [This](https://csmdigital.atlassian.net/browse/CSM-3166) task was the motivator.

### Before submitting this PR, I made sure:

- [x] There is no lint error in the code
- [x] Build process passes successfully
- [x] There are some tests
